components/ws-daemon/pkg/config/config.go

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package config
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"os"
+
+	"github.com/gitpod-io/gitpod/ws-daemon/pkg/daemon"
+	"golang.org/x/xerrors"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+type Config struct {
+	Daemon     daemon.Config `json:"daemon"`
+	Service    AddrTLS       `json:"service"`
+	Prometheus Addr          `json:"prometheus"`
+	PProf      Addr          `json:"pprof"`
+}
+
+type Addr struct {
+	Addr string `json:"address"`
+}
+
+type AddrTLS struct {
+	Addr
```

Comment:
Composition is nice, but here it makes things a tad more complicated. How about:

```Go
type AddrTLS struct {
    Addr string `json:"address"`
    TLS *TLS `json:"tls,omitempty"`
}
```

Reply:
Agreed